document/js-api/index.bs

```diff
@@ -204,6 +204,27 @@ fetch('demo.wasm').then(response =>
 );
 ```
 
+<h2 id="webassembly-storage">Internal storage</h2>
+
+<h3 id="store">Interaction of the WebAssembly Store with JavaScript</h3>
+
+Note: WebAssembly semantics are defined in terms of a [=store=], representing the state of the world, or "memory". WebAssembly operations take a store and return a new store.
+
+Each [=agent=] has an <dfn>associated store</dfn>. When a new agent is created, its associated store is set to the result of [=init_store=]().
+
+Note: In this specification, no WebAssembly-related objects, memory or addresses can be shared among agents in an [=agent cluster=]. In a future version of WebAssembly, the <a href="https://github.com/WebAssembly/threads">threads proposal</a> will add such sharing. For the JS binding, this will likely mean that there will be a store per agent cluster, rather than per agent. This specification uses a store per agent in order to limit semantics to how a store operates in a single-threaded manner, as the WebAssembly specification without threads only deals with that case.
+
+Elements of the WebAssembly store may be <dfn>identified with</dfn> JavaScript values. In particular, each WebAssembly [=memory instance=] with a corresponding {{Memory}} object is identified with a JavaScript [=Data Block=]; modifications to this Data Block are identified to updating the agent's store to a store which reflects those changes, and vice versa.
+
+<h3 id="object-caches">WebAssembly JS Object Caches</h3>
+
+Note: There are several WebAssembly objects that may have a corresponding JavaScript object. The correspondence is stored in a per-agent mapping from WebAssembly addresses to JavaScript objects. This mapping is used to ensure that, for a given [=agent=], there exists at most one JavaScript object for a particular WebAssembly address.
```

Comment:
Link "addresses" to https://webassembly.github.io/spec/core/exec/runtime.html#addresses